lib/inferno/dsl/runnable.rb

```diff
@@ -210,22 +210,24 @@ def handle_child_definition_block(klass, &)
       def id(new_id = nil)
         return @id if new_id.nil? && @id.present?
 
-        prefix =
-          if parent
-            "#{parent.id}-"
-          else
-            ''
-          end
+        prefix = parent ? "#{parent.id}-" : ''
 
         @base_id = new_id || @base_id || default_id
-
         final_id = "#{prefix}#{@base_id}"
+        @full_id = final_id
 
-        raise Exceptions::InvalidRunnableIdException, final_id if final_id.length > 255
+        if final_id.length > 255
+          hash = Digest::SHA1.hexdigest(final_id)[0...10]
+          final_id = "#{final_id[0...244]}-#{hash}"
+        end
 
         @id = final_id
       end
 
+      def full_id
```

Comment:
Rather than storing the hashed id as `id` and the original as `full_id`, I think the original needs to be stored as `id`.

What about something like this?
- an original `id` and a `database_id`
- when setting `id`, `database_id` is set if it is too long
- a `database_id` method returns the `database_id` if it exists, otherwise the `id`
- runnables are stored in their repositories using both their `id` and `database_id`
- the db repositories use `database_id` instead of `id` for runnable associations (it looks like only test runs and results have runnable associations, but it would be nice to have a general solution to this if possible)

Reply:
Not sure if I addressed  the last point "the db repositories use database_id instead of id for runnable associations (it looks like only test runs and results have runnable associations, but it would be nice to have a general solution to this if possible)" as expected.